### ISSUE TITLE
[enterprise-4.4] Edit suggested in file openshift_images/image-configuration.adoc

### ISSUE THREAD

**Issue by pamoedom** [openshift/openshift-docs#22268]
### Which section(s) is the issue in?

https://docs.openshift.com/container-platform/4.4/openshift_images/image-configuration.html#images-configuration-registry-mirror_image-configuration

### What needs fixing?

The example for skopeo copy is invalid because it will not preserve the same Digest if the source-image is a list of manifests, this was fixed since skopeo 0.1.40 with an extra option called "-a":

~~~
   --all, -a                                                       Copy all images if SOURCE-IMAGE is a list
~~~

In summary, the example from the documentation should be as follows instead:

~~~
$ skopeo copy -a \
   docker://registry.access.redhat.com/ubi8/ubi-minimal@sha256:c505667389712dc337986e29ffcb65116879ef27629dc3ce6e1b17727c06e78f \
   docker://example.io/ubi8/ubi-minimal
~~~

OR  (more simpler)

~~~
$ skopeo copy -a \
   docker://registry.access.redhat.com/ubi8/ubi-minimal \
   docker://example.io/ubi8/ubi-minimal
~~~

Best Regards.

**Comment by pamoedom** [comment]
I forgot to add the following:

In the same section is also the following "podman pull" example to demonstrate the mirroring process:

~~~
sh-4.2# podman pull --log-level=debug registry.access.redhat.com/ubi8/ubi-minimal
~~~

However, this command will fail because by default, the clusters implement the option "mirror-by-digest-only = true" within "/etc/containers/registries.conf" for every registry, so the rewrite operation will only suceed if the proper Digest is referenced instead of the tag, for example:

NOTE: If I recall correctly this was added since OCP 4.2.

[TAG FAILS]
~~~
$ podman pull registry.access.redhat.com/ubi8/ubi-minimal --log-level debug

DEBU[0000] reference rewritten from 'registry.access.redhat.com/ubi8/ubi-minimal:latest' to 'registry.access.redhat.com/ubi8/ubi-minimal:latest' 
DEBU[0000] Trying to access "registry.access.redhat.com/ubi8/ubi-minimal:latest"

[DIGEST WORKS]
~~~
$ podman pull registry.access.redhat.com/ubi8/ubi-minimal@sha256:326c94ab44d1472a30d47c49c2f896df687184830fc66a66de00c416885125b0 --log-level debug

DEBU[0000] reference rewritten from 'registry.access.redhat.com/ubi8/ubi-minimal@sha256:326c94ab44d1472a30d47c49c2f896df687184830fc66a66de00c416885125b0' to 'registry.example.com:5000/ubi8/ubi-minimal@sha256:326c94ab44d1472a30d47c49c2f896df687184830fc66a66de00c416885125b0' 
DEBU[0000] reference rewritten from 'registry.access.redhat.com/ubi8/ubi-minimal@sha256:326c94ab44d1472a30d47c49c2f896df687184830fc66a66de00c416885125b0' to 'registry.access.redhat.com/ubi8/ubi-minimal@sha256:326c94ab44d1472a30d47c49c2f896df687184830fc66a66de00c416885125b0' 
DEBU[0000] Trying to access "registry.example.com:5000/ubi8/ubi-minimal@sha256:326c94ab44d1472a30d47c49c2f896df687184830fc66a66de00c416885125b0"
~~~

Best Regards.

**Comment by pamoedom** [comment]
As an extra note, here you have also the confirmation about skopeo "-a" addition within version 0.1.40:

[From https://github.com/containers/skopeo.git]

~~~
commit be6146b0a8471b02e776134119a2c37dfb70d414 (tag: v0.1.40)
Author: Valentin Rothberg <rothberg@redhat.com>
Date:   Mon Oct 28 17:14:34 2019 +0100

    release skopeo v0.1.40
    
    * vendor containers/image v5.0.0
    * copy: add a --all/-a flag
    * System tests: various fixes
    * Temporarily work around auth.json location confusion
    * systemtest: copy: docker->storage->oci-archive
    * systemtest/010-inspect.bats: require only PATH
    * systemtest: add simple env test in inspect.bats
    * bash completion: add comments to keep scattered options in sync
    * bash completion: use read -r instead of disabling SC2207
    * bash completion: support --opt arg completion
    * bash-completion: use replacement instead of sed
    * bash completion: disable shellcheck SC2207
    * bash completion: double-quote to avoid re-splitting
    * bash completions: use bash replacement instead of sed
    * bash completion: remove unused variable
    * bash-completions: split decl and assignment to avoid masking retvals
    * bash completion: double-quote fixes
    * bash completion: hard-set PROG=skopeo
    * bash completion: remove unused variable
    * bash completion: use `||` instead of `-o`
    * bash completion: rm eval on assigned variable
    * copy: add --dest-compress-format and --dest-compress-level
    * flag: add optionalIntValue
    * Makefile: use go proxy
    * inspect --raw: skip the NewImage() step
    * update OCI image-spec to 775207bd45b6cb8153ce218cc59351799217451f
    * inspect.go: inspect env variables
    * ostree: use both image and & storage buildtags
    
    Signed-off-by: Valentin Rothberg <rothberg@redhat.com>
~~~

Regards.
